controlplane/server/internal/registry/registry_sampler.go

```diff
@@ -208,19 +208,22 @@ func (sr *SamplerRegistry) Register(resource string, name string, uid control.Sa
 	return err
 }
 
-func (sr *SamplerRegistry) Deregister(resource string, name string, uid control.SamplerUID) error {
+func (sr *SamplerRegistry) Deregister(uid control.SamplerUID) error {
 	sr.m.Lock()
 	defer sr.m.Unlock()
 
-	sampler, err := sr.getSampler(resource, name)
-	if err != nil {
-		return ErrUnknownSampler
+	// Find sampler instance
+	var found bool
+	var sampler *defs.Sampler
+	for _, sampler = range sr.samplers {
+		_, ok := sampler.GetInstance(uid)
+		if ok {
+			found = true
+			break
+		}
 	}
-
-	_, ok := sampler.Instances[uid]
-	if !ok {
-		sr.logger.Error("deregistering unknown sampler, nothing to do", "sampler_uid", uid)
-		return nil
+	if !found {
+		return ErrUnknownSamplerInstance
```

Comment:
I would keep the log (probably better as a debug log) since this should not happen and may indicate that be have a bug in the protocol

Reply:
Nevermind, at the registry level is fine to always return an error. We will still get the log at the sampler/stream layer.